internal/infrastructure/router/router_test.go

```diff
@@ -151,121 +68,92 @@ func TestLoggerSkipperWithPaths(t *testing.T) {
 
 func TestConfigCommonMiddlewares(t *testing.T) {
 	e := echo.New()
-	rcfg := RouterConfig{
-		MetricsPath: "/metrics",
-		PrivatePath: "",
-	}
-	configCommonMiddlewares(e, rcfg)
+	cfg := config.Get()
+	configCommonMiddlewares(e, cfg)
 }
 
-func TestNewRouterWithConfig(t *testing.T) {
-	assert.Panics(t, func() {
-		NewRouterWithConfig(nil, RouterConfig{})
-	}, "'e' is nil")
+func TestCheckNewRouterWithConfig(t *testing.T) {
+	assert.PanicsWithValue(t, "'e' is nil", func() {
+		NewRouterWithConfig(nil, nil, nil, nil)
+	})
 
 	e := echo.New()
-	assert.Panics(t, func() {
-		NewRouterWithConfig(e, RouterConfig{})
+	assert.PanicsWithValue(t, "'cfg' is nil", func() {
+		NewRouterWithConfig(e, nil, nil, nil)
 	})
 
 	cfg := test.GetTestConfig()
-	reg := prometheus.NewRegistry()
-	metrics := metrics.NewMetrics(reg)
-	_, db, _ := test.NewSqlMock(&gorm.Session{SkipHooks: true})
-	inventory := client_inventory.NewHostInventory(t)
-
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-	svcRbac, mockRbac := mock_rbac.NewRbacMock(ctx, cfg)
-	err := svcRbac.Start()
-	defer svcRbac.Stop()
-	require.NoError(t, err)
-	require.NoError(t, mockRbac.WaitAddress(3*time.Second))
-	mockRbac.SetPermissions(mock_rbac.Profiles["domain-admin-profile"])
-	rbacClient, err := client_rbac.NewClient("idmsvc", client_rbac.WithBaseURL(cfg.Clients.RbacBaseURL))
-	if err != nil {
-		panic(err)
-	}
-	rbac := client_rbac.New(cfg.Clients.RbacBaseURL, rbacClient)
-	require.NotNil(t, rbac)
-	pendo := client_pendo.NewClient(cfg)
-	require.NotNil(t, pendo)
-
-	// Create application handlers
-	app := handler_impl.NewHandler(cfg, db, metrics, inventory, rbac, pendo)
+	assert.PanicsWithValue(t, "'app' is nil", func() {
+		NewRouterWithConfig(e, cfg, nil, nil)
+	})
 
-	goodConfig := RouterConfig{
-		Version:     "1.0",
-		PublicPath:  "/api/idmsvc",
-		PrivatePath: "/private",
-		Handlers:    app,
-		Metrics:     metrics,
-	}
-	badConfig := RouterConfig{}
+	app := handler.NewApplication(t)
+	assert.PanicsWithValue(t, "'metrics' is nil", func() {
+		NewRouterWithConfig(e, cfg, app, nil)
+	})
 
-	assert.Panics(t, func() {
-		_ = NewRouterWithConfig(e, badConfig)
+	reg := prometheus.NewRegistry()
+	metrics := metrics.NewMetrics(reg)
+	assert.NotPanics(t, func() {
+		NewRouterWithConfig(e, cfg, app, metrics)
 	})
+}
+
+func TestNewRouterWithConfig(t *testing.T) {
```

Comment:
Nit: This test seems to cover the same thing as the last part of `TestCheckNewRouterWithConfig`